azure_img_utils/azure_image.py

```diff
@@ -700,6 +722,19 @@ def get_offer_status(self, offer_id: str) -> str:
 
         return 'unkown'
 
+    def wait_on_operation(self, operation_id: str) -> dict:
+        """
+        Wait until the operation finishes then return the dictionary status
+        """
+        while True:
+            operation = self.get_operation(operation_id)
+
+            status = operation.get('jobStatus', 'unknown')
+            if status in ('completed', 'unkown'):
+                return operation
+
+            time.sleep(1)
```

Comment:
How long usually operations take int he API?
Polling each second might be a bit too much? :thinking:

Reply:
Yeah, I would suggest an accelerating wait algorithm. Start with 30 seconds for the first wait for example then wait 15 seconds, 5, 1 and then repeat if we want to give more time. DO we want to wait for days if it takes Azure that long?

Reply:
They are almost instantaneous. Within a second or two, five at most. Basically Azure is just validating the request and updating a document. Why they are async calls 🤷🏼 .